src/couch_peruser/test/couch_peruser_test.erl

```diff
@@ -156,6 +156,20 @@ should_create_user_db_with_default(TestAuthDb) ->
         ?_assertEqual(1, couch_util:get_value(q, ClusterInfo))
     ].
 
+should_create_user_db_with_custom_prefix(TestAuthDb) ->
+    set_config("couch_peruser", "database_prefix", "newuserdb-"),
```

Comment:
This whole test suite is old and flaky. Check out L64 and above it; there is even a `timer:sleep` on L45.

We should be using `config:set()` in the `set_config()` call rather than hitting the `/_config` endpoint to be more sure about configuration changes. We can even then prevent the change from persisting to disk with the 4th argument.

I realize this is technically outside the scope of this PR, but as someone who struggled for 6 months trying to stabilise the test suite, it bugs me that we've put tests in that are more likely to fail on a heavily contended system like Travis or Jenkins...

Reply:
Thanks a lot @wohali. We may use another PR to address the issue you mentioned above.